### PR TITLE
Replace android logcat with `liblog`

### PR DESCRIPTION
**Pull request by yassernasc**

### CODE REVIEW

**Review by kasperisager — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor points! Before we land this we should add Android support to `liblog` as well so we don't lose Logcat integration.

File: CMakeLists.txt [hunk 1]
```diff
@@ -292,6 +280,36 @@ if(target MATCHES "win32")
...
+    bare_shared
+    PRIVATE
+      log_static
+  )
+
+  target_link_libraries(
+    bare_static
+    PRIVATE
+      log_static
+  )
```
[the remark this answers is not in the document]
Comment (yassernasc) [reply CMakeLists.txt:310]: After applying the suggested changes, leaving only:
```js
  target_link_libraries(
    bare
    PUBLIC
      log
  )
```
the `liblog` methods aren't recognized, resulting in `Undefined symbols` errors.

But curiously replacing `log` with `log_static` creates a successful build:
```js
  target_link_libraries(
    bare
    PUBLIC
      log_static
  )
  ```
  
  any guess about how to link `log` correctly?

File: bin/bare.c [hunk 2]
```diff
@@ -18,6 +22,11 @@ main (int argc, char *argv[]) {
   err = bare_setup(uv_default_loop(), platform, NULL, argc, argv, NULL, &bare);
   assert(err == 0);
 
+#if BARE_USE_SYSTEM_LOG
+  err = log_open("bare", 0);
+  assert(err == 0);
+#endif
+
```
Comment (kasperisager) [inline bin/bare.c:29]: Let's move this to the very top of `main()` so other bits can also use the log in the future.

File: bin/bare.c [hunk 3]
```diff
@@ -29,6 +38,11 @@ main (int argc, char *argv[]) {
   err = bare_teardown(bare, &exit_code);
   assert(err == 0);
 
+#if BARE_USE_SYSTEM_LOG
+  err = log_close();
+  assert(err == 0);
+#endif
+
```
Comment (kasperisager) [inline bin/bare.c:45]: Similar to the above, let's move this to the very bottom.

File: CMakeLists.txt [hunk 4]
```diff
@@ -292,6 +280,24 @@ if(target MATCHES "win32")
...
+
+if(BARE_USE_SYSTEM_LOG)
+  if(NOT TARGET log)
+    add_subdirectory(vendor/liblog EXCLUDE_FROM_ALL)
+  endif()
+
+  target_link_libraries(
+    bare
+    PUBLIC
+      log_static
```
Comment (kasperisager) [inline CMakeLists.txt:297]:
```suggestion
      log
```

Linking the object library is preferable here as `bare` is itself an object library.

**Review by kasperisager — CHANGES_REQUESTED** [review]
Just a couple more nits and then it's good to go! 👏

File: CMakeLists.txt (hunk 4 above)
Comment (yassernasc) [inline CMakeLists.txt:297]: That's the trickiest part, by switching `log_static` with `log`, the liblog methods aren't recognized, resulting in cmake `Undefined symbols` errors.

Comment (yassernasc) [reply CMakeLists.txt:297]: Doing so, the liblog methods aren't recognized, resulting in cmake `Undefined symbols` errors:

```
Linking C shared library libbare.dylib
FAILED: libbare.dylib
(...)
Undefined symbols for architecture arm64:
  "_log_error", referenced from:
      _bare_runtime_print_error in runtime.c.o
  "_log_fatal", referenced from:
      _bare_runtime_on_uncaught_exception in runtime.c.o
      _bare_runtime_on_unhandled_rejection in runtime.c.o
  "_log_info", referenced from:
      _bare_runtime_print_info in runtime.c.o
ld: symbol(s) not found for architecture arm64
```

Comment (yassernasc) [reply CMakeLists.txt:297]: Any guess about how to bypass that? I confess that I tried to add `log` to many places like Bare's `target_include_directories` but without success.

Comment (kasperisager) [reply CMakeLists.txt:297]: That's odd, it works without issue on my end 🤔 I'll try pushing the change and see which targets fail.

Comment (kasperisager) [reply CMakeLists.txt]: Arf, that obviously won't show any issues though as we don't compile with the system log enabled. I'll try on my Linux box.

Comment (kasperisager) [reply CMakeLists.txt]: Got it! We weren't including the objects themselves:

```cmake
target_sources(
  bare
  PUBLIC
    $<TARGET_OBJECTS:log>
)
```

**Review by kasperisager — APPROVED** [review]

**Pull request merged**